lib/perl/Genome/Model/Tools/Vcf/Convert/Base.pm

```diff
@@ -389,20 +389,21 @@ sub query_tcga_barcode_error_template {
 
 sub query_tcga_barcode {
     my $self = shift;
-    my $barcode_str = shift;
+    my $barcodes = shift;
     my %argv = @_;
 
-    my $url = delete $argv{url} || 'https://tcga-data.nci.nih.gov/uuid/uuidws/mapping/json/barcode/batch';
     my $sleep = delete $argv{sleep} || 10;
 
     my $agent = LWP::UserAgent->new();
 
     # based on instrumentation the upper bound on response times is 700 ms
     $agent->timeout(5);
 
-    my $request = HTTP::Request->new(POST => $url);
-    $request->content_type('text/plain');
-    $request->content($barcode_str);
+    my $query = join('%2C', map { '%22' . $_ . '%22' } @$barcodes);
+    my $filters = '%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22samples.portions.analytes.aliquots.submitter_id%22%2C%22value%22%3A%5B' . $query. '%5D%7D%7D';
```

Comment:
I might URL encode this on the fly so that the string is a little more readable in code. Likewise with the filters, is it possible to store them in some sort of data structure rather than a string, at least until they're needed?

Basically, is there an OO-ier way of building this query that isn't just a bunch of string concatenation?